### PR TITLE
[REST API] Remove AB test and update flow

### LINKED ISSUE #8321: [REST API] Remove AB test and update login flow

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #8321 

### Description
This PR has two main changes:
1. Removes the AB test.
2. Update the login flow to align with the new plan pe5sF9-17Z-p2, which is basically:
    i. WPCom login for both WPCom and Jetpack sites.
    ii. Site Credentials login for non-Jetpack sites.

##### WPCom site:

1. Open the app (logout if needed).
2. On the prologue screen, notice that there are only 2 buttons: Enter your site address and Get started.
3. Tap Enter your site address and proceed with the address of your WPCom site.
4. Notice that the WPCom email screen is displayed. Proceed to log in with your WPCom credentials.
5. When the login completes, you should be navigated to the home screen.

##### Self-hosted site connected to Jetpack:

1. Open the app (logout if needed).
2. Tap Enter your site address and proceed with the address of your self-hosted site that has Jetpack connected.
3. Notice that the WPCom email screen is displayed. Proceed to log in with your WPCom credentials.
4. When the login completes, you should be navigated to the home screen.
5. All features should be available: site switching, notifications, site visitors on the dashboard...

##### Self-hosted site without Jetpack.

1. Open the app (logout if needed).
2. Tap Enter your site address and proceed with the address of your self-hosted site that doesn't have Jetpack installed, activated or connected to a WPCom account.
3. Notice that the site credential screen is then displayed. Proceed to log in with your site credentials.
4. When the login completes, you should be navigated to the home screen.
5. Notice that a few features are missing:
    - Site visitors on the dashboard.
    - Site switching on the Menu tab.
    - Notifications for new orders or reviews.
    - Password-protected visibility for products.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml [hunk 1]
```diff
@@ -80,17 +80,18 @@
...
-                android:layout_weight="1"
                 android:layout_height="wrap_content"
                 android:layout_marginStart="@dimen/major_100"
                 android:layout_marginEnd="@dimen/minor_100"
                 android:layout_marginBottom="@dimen/minor_50"
+                android:layout_weight="1"
+                android:text="@string/login_wpcom"
                 android:textAllCaps="false"
-                android:text="@string/login_wpcom" />
+                android:visibility="gone" />
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml:90]: I was torn between doing this visibility change and removing the button from the layout, I feel like there is a high chance this change is temporary, and we'll go back to showing it soon, so I decided to go with the visibility for now, but please let me know if you disagree.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/ExperimentTracker.kt [hunk 2]
```diff
@@ -7,8 +7,6 @@ interface ExperimentTracker {
         const val PROLOGUE_EXPERIMENT_ELIGIBLE_EVENT = "prologue_carousel_displayed"
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/ExperimentTracker.kt]: Np unrelated to these changes but I think this constant can be removed too.

File: WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml:90]: I too, have the feeling this button will be back soon 😅

**Comment by JorgeMucientes** [comment]
Hey @hichamboushaba excellent job at cleaning the A/B test and handling the new login low 👏🏼 . 

I just noticed a couple things we should double check. 

- Jetpack uninstalled but Jetpack CP installed for example through WCPayments. The flow works well but I just want to double check that in this cases we want to offer the WP.com login flow. I don't know if you have discussed this somewhere, couldn't see it in P2: 

https://user-images.githubusercontent.com/2663464/218997601-e5f7118e-b971-4846-99d1-95bbeb6d2822.mp4

I honestly don't think we can do much about this scenario. Because when Jetpack Cp is installed the backend returns true for all `jetpack` status fields: 
```
{"is_wp_com":"false","has_jetpack":"true","is_wordpress":"true","login_calculated_has_jetpack":"true","is_jetpack_active":"true","url_after_redirects":"https:\/\/mucitest.mystagingwebsite.com","exists":"true","is_jetpack_connected":"true","url":"https:\/\/mucitest.mystagingwebsite.com\/","is_debug":true}
```

Seems reasonable to go the WP.com login way, even if we then display the install Jetpack banner. 

- I though we addressed this in the past but we have a bug when "clealing" the site url. This leads to inconsistent backend responses. Let's say the site address contains `\wp-admin` path. Then this happens: 

https://user-images.githubusercontent.com/2663464/218998343-b13c5459-8bc6-444c-83ab-1d820ebf740b.mp4

Backend response with `wp-admin` path:
```
login_site_address_site_info_succeeded, Properties: {"is_wp_com":"false","has_jetpack":"true","is_wordpress":"true","login_calculated_has_jetpack":"false","is_jetpack_active":"true","url_after_redirects":"https:\/\/mucitest.mystagingwebsite.com\/wp-login.php","exists":"true","is_jetpack_connected":"false","url":"https:\/\/mucitest.mystagingwebsite.com\/wp-admin\/","is_debug":true}
```

and without it: 

```
login_site_address_site_info_succeeded, Properties: {"is_wp_com":"false","has_jetpack":"true","is_wordpress":"true","login_calculated_has_jetpack":"true","is_jetpack_active":"true","url_after_redirects":"https:\/\/mucitest.mystagingwebsite.com","exists":"true","is_jetpack_connected":"true","url":"https:\/\/mucitest.mystagingwebsite.com","is_debug":true}
```

Imho this shouldn't happen in the backend side, but I think we can fix this in our side by cleaning the url when we check the site status. Doesn't have to be in this PR, so I won't block it because of this.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by hichamboushaba** [comment]
Thanks for the review @JorgeMucientes 

> Jetpack uninstalled but Jetpack CP installed for example through WCPayments.

I'd say this scenario works as expected now, the API was always like this, and it's fine for us, since that's what we need. And in addition to this, connecting using Jetpack CP offers more features than Application Passwords, so this scenario is better also for our users.

> I though we addressed this in the past but we have a bug when "clealing" the site url. This leads to inconsistent backend responses. Let's say the site address contains \wp-admin path. Then this happens:

Nice catch, I'll try to fix in a separate PR, I'll see if it needs a Login library fix or not.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/ExperimentTracker.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/ExperimentTracker.kt]: Done in [b05a01e](https://github.com/woocommerce/woocommerce-android/pull/8323/commits/b05a01e9c7ab2fae50dac1b179ecfecac41d44bf)

**Pull request merged**